src/main/java/net/neoforged/neoforge/common/BooleanAttribute.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common;
+
+import net.minecraft.world.entity.ai.attributes.Attribute;
+import net.minecraft.world.entity.ai.attributes.AttributeModifier;
+
+/**
+ * A boolean attribute only has two states, on or off, represented by a value of 0 (false) or 1 (true).
+ * <p>
+ * For boolean attributes, only use the following modifier values:
```

Comment:
What happens with modifier values outside of these?

Reply:
For addition values > 0, they will work as an enabler, but would be poor form, simply because of the established convention.
For addition values < 0, they would have undefined behavior.  For example, applying a `-2` addition would remove two sources of flight, but if a third was present, would not disable it.

For `ADD_MULTIPLIED_BASE` modifiers, the behavior is also undefined.  Values other than `-1.0` will just do unpredictable things (similar to the addition case above) that will probably not disable flight, and even using `-1.0` may fail to disable flight due to floating-point precision issues.

The only way to appropriately disable flight is `ADD_MULTIPLIED_TOTAL` with `-1.0`, because this forces the calculation `value = value * 0`

Reply:
Hmmmm. Quite interesting.

Should we document these, at least in brief such as "Use of <these> will cause undefined behavior, and therefore should never be used."?

Reply:
added